discv5/crawler.go

```diff
@@ -480,6 +483,8 @@ func (c *Crawler) crawlDiscV5(ctx context.Context, pi PeerInfo) chan DiscV5Resul
 				err = fmt.Errorf("getting closest peer with CPL %d: %w", i, err)
 				break
 			}
+			success = true
```

Comment:
actually, could you use the `errorBits` to determine if there is at least one successful FIND_NODE RPC? Then you wouldn't need the `success` variable.

However, `errorBits` aren't tracked here yet (just saw that), so this also needs fixing. 🙈 Could you also do that? The `drainBuckets` in `crawler_p2p.go` in the `libp2p` package has the logic of how to do that.

Reply:
updated: https://github.com/dennis-tra/nebula/blob/9a1a685b49d5ae259bff1fa603f6461a705d8737/discv5/crawler.go#L516-L521

Reply:
ah sorry, I meant `errorBits` isn't even used in this routine. So it will always be 0 as there's no statement that updates the errorBits variable. In case of an error the corresponding bit should be set (even though it's inaccurate because we might exit the loop early)

Reply:
`errorBits` is updated at
https://github.com/dennis-tra/nebula/blob/9a1a685b49d5ae259bff1fa603f6461a705d8737/discv5/crawler.go#L481

You mean, if we exit the loop early we need to set the error bits to 1 for the buckets that were skipped?

I'll update the `ErrorBits` to `uint32` as it can contain up to `17` bits.

Reply:
lol, I was ctrl+f searching the page for `errorBits` and was afterward expanding the code lines in the GitHub diff viewer. However, this didn't highlight the `errorBits` variable there so I was under the impression that it wasn't used. Sorry for the noise!

Reply:
> I'll update the ErrorBits to uint32 as it can contain up to 17 bits.

`ErrorBits` is defined in `core` and I don't want to mess everything up, so we can assume that bucket 17 (or 239 in discv5 naming) will always be empty and never cause issues 👍🏻